test/e2e/functional_test.go

```diff
@@ -1051,3 +1051,32 @@ spec:
 		})
 
 }
+
+func (s *FunctionalSuite) TestContainerSetRetry() {
+	s.Given().
+		Workflow(`
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  generateName: script-nonroot-
+spec:
+  entrypoint: main
+  templates:
+    - name: main
+      containerSet:
+        containers:
+          - name: a
+            image: docker/whalesay:latest
+            command: [sh, -c]
+            args: ['FILE=/var/run/argo/testy.yml; if test -f "$FILE"; then; EXITCODE=0; else; touch $FILE; EXITCODE=1; fi; local t1=$?;  exit $t1']
```

Comment:
Should leave a file for the second attempt to pass successfully due to that there is no clean.